### ISSUE TITLE
Add Thinning Parameter to Simulation.plot?

### ISSUE THREAD

**Issue by jonstrutz11** [Heuro-labs/cayenne#129]
It would be useful to have a thinning parameter, since when plotting long runs, plotting is time/memory intensive.

For example, one could do something like:

sim.plot(['A', 'B'], thinning=100)

and it would plot every 100 states instead of every single state.

(or maybe this is just something the user should be expected to do with the sim.results themselves, since it is relatively straightforward to implement)

**Comment by jonstrutz11** [comment]
P.S. thanks for such a simple and easy to use KMC interface. I think I'm going to end up using Cayenne for part of my thesis 👍

**Comment by sriki18** [comment]
Thanks for the kind words @jonstrutz11 ! Made my day.

What you are saying absolutely makes sense, and we can add it as soon as this weekend. How do you think we should join the y values between t1 and t100? I am thinking a stair plot will be somewhat inaccurate, but so will a linear interpolation. Maybe end up just plotting the points?

**Comment by jonstrutz11** [comment]
No problem! I searched for days finding a simple-to-use KMC library and this is it, haha. By the way, I tried GillespiePy2 but I realized that it doesn't output exact times of reaction events, it just samples states at preset times (e.g. time = 1, 2, 3, 4, 5 s). Cayenne gives actual times, though, so that is honestly the main reason why I switched >.<

I was thinking just plotting every 100th point for example I mentioned, since that seems the simplest way to do it.

You could do something like take the average of every 100 points, but my intuition is that would result in trajectories that have less stochastic variance than is actually present..

**Comment by jonstrutz11** [comment]
By "join the y values between t1 and t100" do you mean joining them on the plot, e.g. with a line? or do you mean somehow considering the values in between t1 and t100?

**Comment by sriki18** [comment]
Haha I see! I actually had some difficulty getting GillesPy to work in the past...Glad cayenne meets your needs though!

Yes I meant joining it with a line on a plot. The reason to join it with a line is that you can look at the individual trajectories - without the line, you don't know how it got where it is. But we can probably just add that in as an option too

**Comment by jonstrutz11** [comment]
Yep, I think adding a line is probably fine! I imagine any user will be mentioning to others that the data has been thinned so the trajectories are a subset of the true trajectory.

**Comment by sriki18** [comment]
So we decided on this:
1. If thinning is 1, plot the trajectories like normal.
2. Else, plot the trajectories as a scatter plot.

Can you check out the develop branch? You will probably have to install from source, though. Lmk if you need help with that.

https://github.com/Heuro-labs/cayenne/tree/develop

**Comment by jonstrutz11** [comment]
Just installed from source and confirmed it works for me. Thanks so much!
